src/main/java/com/teragrep/cfe_39/consumers/kafka/BatchDistributionImpl.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.cfe_39.consumers.kafka;
+
+import com.google.gson.*;
+import com.teragrep.cfe_39.configuration.ConfigurationImpl;
+import com.teragrep.cfe_39.metrics.topic.TopicCounter;
+import com.teragrep.cfe_39.metrics.DurationStatistics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.codahale.metrics.Timer;
+
+import java.io.*;
+import java.time.Instant;
+import java.util.*;
+
+/*  The kafka stream should first be deserialized using rlo_06 and then serialized again using avro and stored in HDFS.
+  The target where the record is stored in HDFS is based on the topic, partition and offset. ie. topic_name/0.123456 where offset is 123456
+ The mock consumer is activated for testing using the configuration file: readerKafkaProperties.getProperty("useMockKafkaConsumer", "false")*/
+
+public final class BatchDistributionImpl implements BatchDistribution {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BatchDistributionImpl.class);
+
+    private final String topic;
+    private final DurationStatistics durationStatistics;
+    private final TopicCounter topicCounter;
+    private long lastTimeCalled;
+    private final ConfigurationImpl config;
+    private final Map<String, PartitionFileImpl> partitionFileMap;
+
+    public BatchDistributionImpl(
+            ConfigurationImpl config,
+            String topic,
+            DurationStatistics durationStatistics,
+            TopicCounter topicCounter
+    ) {
+        this.config = config;
+        this.topic = topic;
+        this.durationStatistics = durationStatistics;
+        this.topicCounter = topicCounter;
+        this.partitionFileMap = new HashMap<>();
+        this.lastTimeCalled = Instant.now().toEpochMilli();
+    }
+
+    /* Input parameter is a batch of RecordOffsetObjects from kafka. Each object contains a record and its metadata (topic, partition and offset).
+    * Distributes the received kafka record batch to PartitionFileImpl objects based on topic partition which the record originates from.
+    * */
+    @Override
+    public void accept(List<KafkaRecordImpl> batch) {
+        long thisTime = Instant.now().toEpochMilli();
+        long ftook = thisTime - lastTimeCalled;
+        topicCounter.setKafkaLatency(ftook);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER
+                    .debug(
+                            "Fuura searching your batch for <[{}]> with records <{}> and took  <{}> milliseconds. <{}> EPS. ",
```

Comment:
perhaps we should drop the joke about Fuura and have a proper message now?

Reply:
Solved in commit 7a21cda46c45a51bf44f321a1d95ae7164f278c7